src/main/webapp/script.js

```diff
@@ -60,6 +60,27 @@ function passQuery() {
     });
 }
 
+function validateLocation() {
+  const datalist = document.getElementById('location');
+  const inputlist = document.getElementById('locationlist');
+
+  const options = datalist.options;
+  const typedSoFar = inputlist.value.toLowerCase();
+  let matchesAnOption = false;
+
+  for (const option of options) {
+    if (option.value.toLowerCase().includes(typedSoFar)) {
+      matchesAnOption = true;
+      inputlist.style.borderColor = '#767676'; // dark gray
```

Comment:
a cleaner option is to defne css classes for invalid-input and valid-input and here you would just change the class to one or the other. This way you later change how you style invalid and valid input just by changing css and not touching any code.

Reply:
Done